---
title: "Figure 4 and related supplement tables and figures"
output: html_notebook
author: "Meng-Ching Ko (MaggieMCKO)"
date: "06/23/2022"
---

This [R Markdown](http://rmarkdown.rstudio.com) Notebook contain codes for reproducing Fig.4 and related supplement tables and figures of Ko et al. (<https://www.biorxiv.org/content/10.1101/2022.06.13.495861v1>).
Data deposited on dryad: https://doi.org/10.5061/dryad.5hqbzkh8c

### load data
```{r}
library(tidyverse) # v.1.3.1
library(RColorBrewer) # 1.1-3
library(WGCNA); # 1.71

set.seed(100)

# trait 
path = paste0(getwd(), "/Data/WGCNA/Traits.tsv")
datTraits = read_tsv(path)
datTraits0 = datTraits[, -1] %>% as.matrix()
row.names(datTraits0) = datTraits$sample

# exp 
path = paste0(getwd(), "/Data/WGCNA/Expression_est_perBird.tsv")
datExprori = read_tsv(path)
datExpr0 = datExprori[, -1] %>% as.matrix()
row.names(datExpr0) = datExprori$sample
```

### filter samples
```{r}
gsg = goodSamplesGenes(datExpr0, verbose = 3);
gsg$allOK
#If the last statement returns TRUE, all genes have passed the cuts. If not, we remove the oending genes and samples from the data:
if (!gsg$allOK)
{
  # Optionally, print the gene and sample names that were removed:
  if (sum(!gsg$goodGenes)>0)
    printFlush(paste("Removing genes:", paste(names(datExpr0)[!gsg$goodGenes], collapse = ", ")));
  if (sum(!gsg$goodSamples)>0)
    printFlush(paste("Removing samples:", paste(rownames(datExpr0)[!gsg$goodSamples], collapse = ", ")));
  # Remove the offending genes and samples from the data:
  datExpr0 = datExpr0[gsg$goodSamples, gsg$goodGenes]
}

## smaple tree clustering 
# cluster the samples (in contrast to clustering genes that will come later) to see if there are any obvious outliers. 
sampleTree = hclust(dist(datExpr0), method = "average");
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", 
     cex.lab = 1, cex.axis = 1, cex.main = 1)

# choosing samples
MEDissThres = 60
# Determine cluster under the line
clust = cutreeStatic(sampleTree, cutHeight = MEDissThres, minSize = 1)
table(clust)

# clust 1 contains the samples we want to keep.
keepSamples = (clust==1)
deletSamples = row.names(datExpr0)[clust == 2]; deletSamples
# "SDf_T3h_H_764R"

# remove traits from the outlier
ind = which(row.names(datTraits0)%in% deletSamples); ind
datTraits0 = datTraits0[-ind, ]

# make a new expression data frame
datExpr = datExpr0[keepSamples, ]
nGenes = ncol(datExpr); nGenes     # 12360
nSamples = nrow(datExpr); nSamples # 39
rm(datExpr0)

## match trait data =
collectGarbage();

# Re-cluster samples
sampleTree2 = hclust(dist(datExpr), method = "average")

# Convert traits to a color representation: white means low, red means high, grey means missing entry
traitColors = numbers2colors(datTraits0, signed = FALSE);

# Plot the sample dendrogram and the colors underneath.
plotDendroAndColors(sampleTree2, traitColors,
                    groupLabels = colnames(datTraits0),
                    marAll = c(1, 8, 2, 1), # addGuide = FALSE, setLayout = FALSE,
                    cex.dendroLabels = 0.7, 
                    cex.colorLabels = 0.7,
                    main = "Sample dendrogram and trait heatmap")
```

### Network construction and module detection 
```{r}
## Choose a set of soft-thresholding powers
networktype = "signed hybrid"
powers = c(c(1:12), seq(from = 14, to=20, by=2))

# Call the network topology analysis function
sft = pickSoftThreshold(datExpr, dataIsExpr = TRUE, powerVector = powers, networkType = networktype, verbose = 5) 

# Plot the results:
plot.new()
par(mfrow = c(1,2));
cex1 = 1;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")
abline(h=0.85,col="red")
abline(h=0.80,col="red")
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
#  choose the power value which is the lowest power that reaches high "Scale Free Topology Model Fit,signed R^2"


## module manual calculation 
# Co-expression similarity and adjacency
softPower = 6 ; 
adjacency = adjacency(datExpr, power = softPower, type = networktype)

# export to cytoscape
# cytoscape = exportNetworkToCytoscape(adjacency,
#                          edgeFile = paste0(getwd(), "/Data/WGCNA/edge.txt"),
#                          nodeFile = paste0(getwd(), "/Data/WGCNA/node.txt"))
```

### Topological Overlap Matrix (TOM)
```{r}
# Turn adjacency into topological overlap
collectGarbage()
TOM = TOMsimilarity(adjacency, TOMType = "unsigned");
dissTOM = 1-TOM
rm(TOM)


## Clustering using TOM
# Call the hierarchical clustering function
geneTree = hclust(as.dist(dissTOM), method = "average");
# Plot the resulting clustering tree (dendrogram)
plot(geneTree, xlab="", sub="", main = "Gene clustering on TOM-based dissimilarity",
     labels = FALSE, hang = 0.04);

## Module identification using dynamic tree cut:
minModuleSize = 100;
deepSplit = 4;
dynamicMods = cutreeDynamic(dendro = geneTree, distM = dissTOM, method = "hybrid", 
                            deepSplit = deepSplit, pamRespectsDendro = FALSE,
                            minClusterSize = minModuleSize);

## Convert numeric lables into colors
# to remove color that is too light
color_to_remove = c(8, 16, 19, 27, 31, 41:44, 50, 51, 58, 87, 90, 115, 119)
colorSeq = labels2colors(setdiff(1:(length(table(dynamicMods))+length(color_to_remove)), color_to_remove))

dynamicColors = labels2colors(dynamicMods, colorSeq = colorSeq)
table(dynamicColors)

# Plot the dendrogram and colors underneath
plotDendroAndColors(geneTree, dynamicColors, "Dynamic Tree Cut",
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05,
                    main = "Gene dendrogram and module colors")
```

###  Merging of modules whose expression profiles are very similar
```{r}
# Calculate eigengenes
MEList = moduleEigengenes(datExpr, colors = dynamicColors)
MEs = MEList$eigengenes

# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);

# Cluster module eigengenes
METree = hclust(as.dist(MEDiss), method = "average");

# Plot the result
plot(METree, main = "Clustering of module eigengenes",
     xlab = "", sub = "")

# We choose a height cut of 0.2, corresponding to correlation of 0.8, to merge
MEDissThres = 0.2
# Plot the cut line into the dendrogram
abline(h=MEDissThres, col = "red")

plot.new()
par(cex = 1, cex.axis= 1, cex.main = 1.1, cex.lab = 1);
par(mar=c(0, 5, 2, 0)) # c(bottom, left, top, right)
par(mgp=c(2.5, 1,.5)) # axis title, axis labels and axis line.
plot(METree, main = "Clustering of module eigengenes_SDf",
     xlab = "", sub = "")
abline(h=MEDissThres, col = "red")

# Call an automatic merging function
merge = mergeCloseModules(datExpr, dynamicColors, cutHeight = MEDissThres, verbose = 3)
# The merged module colors
mergedColors = merge$colors;
# Eigengenes of the new merged modules:
mergedMEs = merge$newMEs
table(mergedColors)
length(unique(mergedColors))

plot.new()
par(mgp=c(2.5, 1,.5)) # axis title, axis labels and axis line.
plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors),
                    c("Dynamic Tree Cut", "Merged dynamic"),
                    dendroLabels = FALSE, hang = 0.03, marAll = c(1, 6, 3, 1), 
                    addGuide = TRUE, guideHang = 0.05)


# Rename to mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", standardColors(50)); 
moduleLabels = match(mergedColors, colorOrder)-1;
MEs = mergedMEs

# Recalculate MEs with color labels 
nGenes = ncol(datExpr); nGenes # transcript:26061; gene: 12360
nSamples = nrow(datExpr); nSamples # 39

# Calculates module eigengenes 
MEs0 = moduleEigengenes(datExpr, mergedColors)$eigengenes
MEs = orderMEs(MEs0)

# Put close eigenvectors next to each other
MET=orderMEs(cbind(MEs, datTraits0)) 
# Plot the relationships among the eigengenes and the trait 
plot.new()
par(cex = 1, cex.axis= 1, cex.main = 1.1, cex.lab = 1);
plotEigengeneNetworks(MET,"", marDendro=c(0,11,1,5.5), marHeatmap=c(10,12,.5,1.5), 
                      cex.lab = 1, xLabelsAngle=90, printPreservation=T)

```

### Module-trait relationships
```{r}

datTraits_tmp = datTraits0
datTraits_tmp1_6 = datTraits_tmp[,1:6]
datTraits_tmp7 = datTraits_tmp[,7:ncol(datTraits_tmp)]
datTraits_tmp7[is.na(datTraits_tmp7)]  = 0 
# ind_bird2 = grep(788, row.names(datTraits_tmp7))
# datTraits_tmp7[ind_bird2, 1] = 0.01529500 # songrate (not songrate_2s)
datTraits_tmp = cbind(datTraits_tmp1_6, datTraits_tmp7)


# reorder MEs
MEs0 = MEs
names(MEs0)
sel = c("MEgrey60", "MEbrown", 
        "MEblue", "MEturquoise", "MEgreen", "MEred", 
        "MEblack", "MEcyan", "MEyellow", "MEgrey", "MEmagenta",
        "MEsalmon", "MEmidnightblue", "MEgreenyellow"
)
ind_c = sapply(sel, function(s){grep(paste0(s, "$"), names(MEs0))})
MEs = MEs0[, ind_c]

datTraits_tmp_0 = datTraits_tmp
sel = c("MedianSlope", "IqrSlope")
ind_c = sapply(sel, function(s){grep(paste0(s, "$"), colnames(datTraits_tmp_0))})
ind_c_r = c( setdiff(1:ncol(datTraits_tmp_0), ind_c), ind_c)
datTraits_tmp = datTraits_tmp_0[, ind_c_r]

moduleTraitCor0 = cor(MEs, datTraits_tmp, use = "p");
moduleTraitPvalue0 = corPvalueStudent(moduleTraitCor0, nSamples);
moduleTraitPvalue.Adj0 = apply(moduleTraitPvalue0, MARGIN = 2, FUN = function(s){p.adjust(s, "bonferroni")})
uniqModuleCol0 = substring(row.names(moduleTraitCor0), 3)
ModuleCount0 = sapply(1:length(uniqModuleCol0), FUN = function(i){
  mduleI = uniqModuleCol0[i];
  signif(length(which(mergedColors == mduleI)), 3)})
textMatrix0 = paste(signif(moduleTraitCor0, 2), "\n(", signif(moduleTraitPvalue0, 1), ")", sep = "");
dim(textMatrix0) = dim(moduleTraitCor0)
ind_non_sig0 = moduleTraitPvalue0 > 0.05
textMatrix0[ind_non_sig0] = ""

```

#### Fig.4A plot
```{r}
plot.new()
par(mar = c(7.5, 12, 3, 1), cex = 0.7);
# Display the correlation values within a heatmap plot
labeledHeatmap(Matrix = moduleTraitCor0,
               xLabels = colnames(datTraits_tmp),
               yLabels = names(MEs),
               ySymbols = paste(names(MEs), ": ", ModuleCount0, sep=""),
               # ySymbols = names(MEs),
               colorLabels = FALSE,
               colors = blueWhiteRed(50),
               textMatrix = textMatrix0,
               setStdMargins = FALSE,
               cex.text = 0.7,
               cex.lab = 1,
               zlim = c(-1,1),
               main = paste("Module-trait relationships"))

```

### geneTraitSignificance 
#### Fig. 4 - Figure supplement 1A
```{r}
geneTraitSignificance0 = as.data.frame(cor(datExpr, datTraits_tmp, use = "p"));
GSPvalue0 = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance0), nSamples));
GSPvalue0.Adj = apply(GSPvalue0, MARGIN = 2, FUN = function(s){p.adjust(s, "fdr")})
names(geneTraitSignificance0) = paste0("GS.", colnames(datTraits_tmp));
names(GSPvalue0) = paste0("p.GS.", colnames(datTraits_tmp));
names(GSPvalue0.Adj) = paste0("padj.GS.", colnames(datTraits_tmp));
dim(geneTraitSignificance0) # 12360    24
dim(GSPvalue0.Adj)          # 12360    24

# geneTraitSignificance 
GS_dataframe = sapply(1:ncol(geneTraitSignificance0), function(i){
  c(geneTraitSignificance0[, i], GSPvalue0[, i], GSPvalue0.Adj[, i])
})
GS_dataframe = as.data.frame(t(matrix(GS_dataframe, ncol = nrow(geneTraitSignificance0), byrow = T)))
dim(GS_dataframe) #  12360    72
names(GS_dataframe) = paste0(c("GS.", "p.GS", "padj.GS"),  
                             rep(colnames(datTraits_tmp), each= 3))
row.names(GS_dataframe) = colnames(datExpr)
GS_dataframe = cbind(moduleColor = mergedColors, GS_dataframe)

# plot
plot.new()
par(cex = .5, cex.axis=.5, cex.main = 1, cex.lab = .5);
par(mgp=c(2.5, 1,.5)) # axis title, axis labels and axis line.
plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors, numbers2colors(geneTraitSignificance0, signed = TRUE)),
                    c("Dynamic Tree Cut", "Merged dynamic", names(geneTraitSignificance0)),
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = F, guideHang = 0.05, marAll = c(0, 6, 2, 0))
```
### geneModuleMembership 
```{r}
modNames = substring(names(MEs), 3)
geneModuleMembership = cor(datExpr, MEs, use = "p");
MMPvalue = corPvalueStudent(as.matrix(geneModuleMembership), nSamples)
MMPvalue.Adj = apply(MMPvalue, MARGIN = 2, FUN = function(s){p.adjust(s, "fdr")})
names(geneModuleMembership) = paste0("MM", modNames);
names(MMPvalue) = paste0("p.MM", modNames);
names(MMPvalue.Adj) = paste0("padj.MM", modNames);

## geneModuleMembership 
dim(geneModuleMembership) # 12360    13
dim(MMPvalue)             # 12360    13
MM_dataframe = sapply(1:ncol(geneModuleMembership), function(i){
  c(geneModuleMembership[, i], MMPvalue[, i], MMPvalue.Adj[, i])
})
MM_dataframe = as.data.frame(t(matrix(MM_dataframe, ncol = nrow(geneModuleMembership), byrow = T)))
dim(MM_dataframe) #  12360    39
names(MM_dataframe) = paste0(c("MM", "p.MM", "padj.MM"),  rep(modNames, each = 3))
row.names(MM_dataframe) = colnames(datExpr)
MM_dataframe = cbind(moduleColor = mergedColors, MM_dataframe)

# plot
plot.new()
par(cex = .8, cex.axis=.8, cex.main = 1, cex.lab = .8);
par(mgp=c(2.5, 1,.5)) # axis title, axis labels and axis line.
plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors, numbers2colors(geneModuleMembership, signed = TRUE)),
                    c("Dynamic Tree Cut", "Merged dynamic", names(geneModuleMembership)),
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = F, guideHang = 0.05, marAll = c(0, 6, 2, 0))

```

### module data 
#### Supplementary Table 10
```{r}
intramodularConnectivity = intramodularConnectivity(adjacency, mergedColors, scaleByMax = FALSE)
geneInfo0 = data.frame(GeneSymbol = colnames(datExpr),
                       moduleColor = mergedColors,
                       intramodularConnectivity,
                       geneTraitSignificance0)
```


#### Fig. 4 - Figure supplement 1B-E
```{r}
library(grid) # 4.1.2
library(gridExtra) # 2.3

theme_m = theme_classic() + 
  theme(legend.position = 'none', plot.title = element_text(hjust = 0.5), 
        axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title.x = element_blank())

AveConnectivity = geneInfo0 %>%
  group_by(moduleColor) %>%
  summarise(Mean_ktotal = mean(kTotal))
ind_sort = sort(AveConnectivity$Mean_ktotal, index.return = TRUE, decreasing = TRUE)
Module_col = AveConnectivity$moduleColor[ind_sort$ix]; Module_col
AveConnectivity = within(AveConnectivity, moduleColor <- factor(moduleColor, levels = Module_col))
Mean_ktotal_p = ggplot(AveConnectivity, aes(x = moduleColor, y = Mean_ktotal, col = moduleColor)) +
  geom_hline(yintercept = 0, color = "darkgrey", size = 0.2) +
  geom_point() + 
  scale_y_continuous(name = "Mean connectivity") +
  scale_color_manual(values = Module_col) +
  ggtitle("Overall") +
  theme_m ; Mean_ktotal_p

AveConnectivity = geneInfo0 %>%
  group_by(moduleColor) %>%
  summarise(Mean_kDiff = mean(kDiff))
ind_sort = sort(AveConnectivity$Mean_kDiff, index.return = TRUE, decreasing = TRUE)
Module_col = AveConnectivity$moduleColor[ind_sort$ix]; Module_col
AveConnectivity$moduleColor <- factor(AveConnectivity$moduleColor, levels = Module_col)
Mean_kDiff_p = ggplot(AveConnectivity, aes(x = moduleColor, y = Mean_kDiff, col = moduleColor)) +
  geom_hline(yintercept = 0, color = "darkgrey", size = 0.2) +
  geom_point() + 
  scale_y_continuous(name = "Mean connectivity") +
  scale_color_manual(values = Module_col) +
  ggtitle("Differential\n(Intra - Inter)") +
  theme_m ; Mean_kDiff_p

AveConnectivity = geneInfo0 %>%
  group_by(moduleColor) %>%
  summarise(Mean_kWithin = mean(kWithin))
ind_sort = sort(AveConnectivity$Mean_kWithin, index.return = TRUE, decreasing = TRUE)
Module_col = AveConnectivity$moduleColor[ind_sort$ix]; Module_col
AveConnectivity$moduleColor <- factor(AveConnectivity$moduleColor, levels = Module_col)
Mean_kWithin_p = ggplot(AveConnectivity, aes(x = moduleColor, y = Mean_kWithin, col = moduleColor)) +
  geom_hline(yintercept = 0, color = "darkgrey", size = 0.2) +
  geom_point() + 
  scale_y_continuous(name = "Mean connectivity") +
  scale_color_manual(values = Module_col) +
  ggtitle("Intramodular") +
  theme_m ; Mean_kWithin_p

AveConnectivity = geneInfo0 %>%
  group_by(moduleColor) %>%
  summarise(Mean_kOut = mean(kOut))
ind_sort = sort(AveConnectivity$Mean_kOut, index.return = TRUE, decreasing = TRUE)
Module_col = AveConnectivity$moduleColor[ind_sort$ix]; Module_col
AveConnectivity$moduleColor <- factor(AveConnectivity$moduleColor, levels = Module_col)
Mean_kOut_p = ggplot(AveConnectivity, aes(x = moduleColor, y = Mean_kOut, col = moduleColor)) +
  geom_hline(yintercept = 0, color = "darkgrey", size = 0.2) +
  geom_point() + 
  scale_y_continuous(name = "Mean connectivity") +
  scale_color_manual(values = Module_col) +
  ggtitle("Intermodular") +
  theme_m ; Mean_kOut_p

p = arrangeGrob(grobs = list(Mean_ktotal_p, Mean_kDiff_p, Mean_kWithin_p, Mean_kOut_p), ncol = 4)
grid.draw(p)
```

### Cell Type
```{r}
library(tidyverse) # v.1.3.1

path =  paste0(getwd(), "/Data/WGCNA/Modules.tsv")
geneInfo0 = read_tsv(path)
allmodNames = unique(geneInfo0$moduleColor)

path = paste0(getwd(), "/Data/CellType_GeneList.tsv")
cell_type_gene_lists_df = read_tsv(path)

cell_typeS = c( 
  # vessel
  "Endothelial cell", "Fibroblast", "Pericyte", "Smooth muscle cell",
  # Glia
  "Glial", "Astrocyte", "Microglia", "Oligodendrocyte", 
  # neuron
  "Neuron", "Neuronal", "Interneuron", "Projection neuron", 
  # neuronal process
  "Axon", "Axon hillock", "Myelin", "Dendrite", "Spine" , "Gap junction", 
  # neurotransmitters
  "Adrenergic receptor", "Epinephrine", "Norepinephrine", 
  "Dopamine receptor", "Dopamine", "GABA receptor", "GABA", 
  "Glutamate receptor", "Glutamate", "Glycine receptor", "Glycine", 
  "Histamine receptor", "Histamine", "Serotonin receptor", "Serotonin",
  # Genesis
  "Neural stem cell", "Radial glial cell")

## expectation standardize to expectation value [function] 
Plot_gene_expectationRatio = function(temp_cell_type, padj, return_plot){
  # temp_cell_type = cell_typeS[1]
  # padj = 'fdr'
  
  print(temp_cell_type)
  
  temp_cell_type_list = cell_type_gene_lists_df %>% filter(CellType == temp_cell_type)
  
  temp_intersectS = sapply(X = 1:length(allmodNames), FUN = function(j){
    print(paste0("j = ", j))
    
    ## here
    temp_updw = geneInfo0 %>% filter(moduleColor == allmodNames[j]) %>% pull(GeneSymbol)
    
    temp = intersect(temp_updw, temp_cell_type_list$GeneSymbol) ;
    count = length(temp);
    len_timepoint = length(unique(temp_updw)) ;
    len_cell_list = length(unique(temp_cell_type_list$GeneSymbol)) ;
    len_universe = 15609 # (protein-coding)
    # canary genome: https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Serinus_canaria/101/#FeatureCountsStats;
    expect = len_timepoint*len_cell_list/len_universe ;
    mat <- matrix(
      c(count,                       # gene in list & in the interested category
        len_timepoint - count,       # gene in list but in the interested category
        len_cell_list,               # gene in universe & in the interested category
        len_universe - len_cell_list # gene in universe but not in the interested category
        ),  
     nrow = 2);
    fr <- fisher.test(mat, alternative = "greater")
    # print(expect)
    return(c(count, expect, signif(fr$p.value, 10) )) 
    })
  
  temp_intersectS = as.data.frame(t(temp_intersectS));
  names(temp_intersectS) = c("Gene count", "Expectation", "P-value");
  temp_intersectS$Module = allmodNames;
  temp_intersectS$`Search term` = temp_cell_type;
  temp_intersectS$Substraction = temp_intersectS$`Gene count`- temp_intersectS$Expectation;
  temp_intersectS$Ratio = temp_intersectS$Substraction/temp_intersectS$Expectation*100
  temp_intersectS$FDR = p.adjust(temp_intersectS$`P-value`, method = padj) # bonferroni, fdr
  temp_intersectS$Significance = ifelse(temp_intersectS$FDR < 0.05, '*', NA) # bonferroni, fdr
  temp_intersectS = within(temp_intersectS, Module <- factor(Module, levels = allmodNames))
  
  if(return_plot == TRUE){
    ggplot(temp_intersectS, aes(x = Module, y = Ratio, fill = Module)) +
      geom_col() +
      geom_hline(yintercept = 0, color = "darkgrey", size = 0.25) +
      geom_text(data = subset(temp_intersectS, su = FDR < 0.05), label = "*") +
      scale_fill_manual(values = allmodNames) +
      scale_y_continuous(name = "Enrichment\nscore") +
      ggtitle(paste(temp_cell_type, dir_title, sep = "\n")) +
      theme_m + 
      theme(plot.title = element_text(hjust = 0.5),
            axis.text.x = element_blank(),
            axis.ticks.x = element_blank()
            ) ;
  }else{temp_intersectS}
}

```


#### Supplementary Table 11
```{r}
multiple_correction_method = "fdr"
plot_out = mapply(cell_typeS, 
                  padj = multiple_correction_method, 
                  return_plot = FALSE,
                  FUN = Plot_gene_expectationRatio, SIMPLIFY = FALSE)
plot_out_r = plot_out %>% bind_rows()

# reorganize table
plot_out_r2 = plot_out_r %>% 
  select(`Search term`, Module, `Gene count`, Expectation, `P-value`, FDR, Significance)
```

#### Fig.4B plot
```{r}
rbind_df = plot_out_r2 %>% mutate(`adjusted p value` = ifelse(FDR > 0.05, NA, FDR))

#### Cell types
rbind_df_1 = rbind_df %>% 
  filter(`Search term` %in%  c( "Oligodendrocyte", "Projection neuron"))

#### Neuron location + neurontransmitter
rbind_df_2 = rbind_df %>% 
  filter(`Search term` %in% c("Axon", "Myelin", "Dendrite", "Spine"))

#### combine both
rbind_df = rbind(cbind("Source" = "Cell type", rbind_df_1),
                 cbind("Source" = "Neuronal", rbind_df_2)) 

rbind_df$`Search term` <- factor(rbind_df$`Search term`, levels = c("Oligodendrocyte", "Projection neuron",
                                                          "Axon", "Myelin", "Dendrite", "Spine"))
rbind_df$Module <- factor(rbind_df$Module, 
                          levels = rev(c("blue", "turquoise", "green", "red", 
                                         "grey60", "midnightblue", "salmon", 
                                         "brown", "greenyellow", "black", "cyan", 
                                         "magenta", "yellow", "grey")))

sig_col = 'darkorange'
high_col = brewer.pal(n = 9, "Oranges")[1]

NeuronLoc_p = ggplot(rbind_df, aes(y = Module, x = `Search term`, fill = `adjusted p value`)) +
  geom_tile(color = "grey90", size = 0.01) +
  scale_fill_continuous(high = high_col, low = sig_col, na.value = alpha("grey90", 0.5),
                        breaks = c(0.01, 0.025, 0.04)) + 
  scale_y_discrete(expand = c(0,0)) +
  scale_x_discrete(expand = c(0,0)) + 
  facet_grid( .~ Source , scales = "free", space = "free") + # subset
theme_classic() + 
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.x = element_text(hjust = 1, angle = 45),
        axis.text.y = element_text(hjust = 1)); NeuronLoc_p
```

### Fig.4C: prep inputs needed for circos 
```{r}
library(tidyverse) # v.1.3.1

# load node file
path =  paste0(getwd(), "/Data/WGCNA/Modules.tsv")
geneInfo0 = read_tsv(path)
node = geneInfo0 %>% dplyr::select(GeneSymbol, moduleColor, kWithin)

# load edge file
path = paste0(getwd(), "/Data/WGCNA/edge.txt") # 335845
edge = read_tsv(path)[, 1:3] 

## make karyotype
node_m = node %>% arrange(moduleColor, desc(kWithin))
sort_col = sort(table(node_m$moduleColor), decreasing = TRUE)

chr = paste0("chr - ", names(sort_col))
id = 1:length(chr)
zero = rep(0, length(chr))
cord = sort_col

karyo = data.frame(chr,  zero, cord, id)
ind = sapply(c("chr", "Var1", "zero", "Freq", "id"), function(s){
  grep(s, names(karyo))
})
karyo = karyo[, ind]
# write_tsv(karyo, file = paste0(getwd(), "/Data/circos/module_karyotype.txt"), col_names = F) 


## make data files
cord_end = lapply(names(sort_col), function(s){
  indd = which(node_m$moduleColor == s);
  df = arrange(node_m[indd, ], desc(kWithin));
  df$end = 1:length(indd)
  return(df)
})
node_m2 = cord_end %>% bind_rows()
node_m2$start = node_m2$end-1

# node
names(node_m2)
ind = sapply(c("moduleColor", "start", "end", "GeneSymbol"), function(s){
  grep(s, names(node_m2))
})
cord_out = node_m2[, ind, with = F]
# write_tsv(cord_out, file = paste0(getwd(), "/Data/circos/cordinates.txt"), col_names = F) 

## kWithin
ind = sapply(c("moduleColor", "start", "end", "kWithin"), function(s){
  grep(s, names(node_m2))
})
kWithin_out = node_m2[, ind, with = F]
# write_tsv(kWithin_out, file = paste0(getwd(), "/Data/circos/kWithin.txt"), col_names = F)

## edge
ind = sapply(c("moduleColor", "start", "end"), function(s){
  grep(s, names(node_m2))
})
line_out = node_m2[, ind, with = F]
line_out$y = 1
# write_tsv(line_out, file = paste0(getwd(), "/Data/circos/line_out.txt"), col_names = F)

## make text label
target = c("SP8", "ESR1", "ESR2", "AR", "GATA2", "GATA3", "GATA4", "MYBL1", "SIN3A", 
           "BDNF", "VEGFA", "VEGFC", "KDR", "FLT4", "FLT1")

sapply(target, function(s){ind = which(node$GeneSymbol == s); node$GeneSymbol[unique(ind)]})
sapply(target, function(s){ind = grep(s, node$GeneSymbol); node$GeneSymbol[unique(ind)]})

text_lab = subset(cord_out, su = GeneSymbol %in% target)
# write_tsv(text_lab, file = paste0(getwd(), "/Data/circos/text_labels_0726.txt"), col_names = F)


## make link file
link_from = cord_out[match(edge$fromNode, cord_out$GeneSymbol),]
link_to   = cord_out[match(edge$toNode, cord_out$GeneSymbol),]
link = cbind(link_from, link_to)
names(link)
ind = sapply(c("GeneSymbol"), function(s){
  grep(s, names(link))
})
link = link[, -ind, with = F]
link = as.data.frame(link)

## binary color SP8 nodes (more color) vs non-SP8node
link$para = paste0("color=", "vlgrey", "_a3") # a1(16%)~a5(83%) for alpha

ind_from_sp8 = which(edge$fromNode == "SP8")
ind_to_sp8 = which(edge$toNode == "SP8")
ind_sp8 = c(ind_from_sp8, ind_to_sp8)

color_blues = c("vvlblue", "vlblue", "blue") # the color defined in circos
color_blues_2 = rep(color_blues, length.out = length(ind_sp8))
link$para[ind_sp8] = paste0("color=", color_blues_2, "_a3")

link = arrange(link, desc(para))
# write_tsv(link, file = paste0(getwd(), "/Data/circos/link_sp8_blues3_vlgrey_0726.txt"), col_names = F)


## for circos.conf
# chromosomes_color
paste0("/", names(sort_col), "/:", names(sort_col), collapse = ";")
# link color
paste0("/", names(sort_col), "/:", names(sort_col), "_a4", collapse = ";")
# color def
col_def = sapply(names(sort_col), function(s){
  paste0(s, " = ", paste0(col2rgb(s), collapse = ","))
})
cat(col_def, sep = "\n")

# then run circos-0.69-4  ## http://circos.ca
# circos -conf ./circos.conf 
```

### Fig. 4 - Figure supplement 2
```{r}
library(tidyverse) # v.1.3.1
library(HiveR) # 0.3.63

# load string output
path = paste0(getwd(), "/Data/WGCNA/SP8_string_nodes_20180531.tsv") 
node = read_tsv(path)

# select modulecolor for axises
unique(node$regulatory.function)
sort(table(node$regulatory.function))
node$regulatory.function = gsub("TF_co", "TF", node$regulatory.function)
sel_col = c("TF", "kinase", "actin")
node_sub = subset(node, su = regulatory.function %in% sel_col, se = c("regulatory.function", "gene.symbol"))
unique(node_sub$regulatory.function)

# load edge file
path = paste0(getwd(), "/Data/WGCNA/string_interactions_tur_adjwithSP8_1151_ZF.tsv")
edge = read_tsv(path)

# subset edges according to node
## only direct neighbors
edge_sub = subset(edge, su = node1 %in% c("SP8") | node2 %in% c("SP8"))
all_edges_sub_node = unique(c(edge_sub$node1, edge_sub$node2))
node_sub = subset(node_sub, su = gene.symbol %in% all_edges_sub_node)

# check how many module colors are left
all_edges_sub_node = unique(c(edge_sub$node1, edge_sub$node2))
unique(node_sub$regulatory.function[which(node_sub$gene.symbol %in% all_edges_sub_node)])

## manuelly making NODE
# lab
names(node_sub) = gsub("gene.symbol", "lab", names(node_sub))
# col
df = data.frame(regulatory.function = sel_col, color = c("#FF9900", "#00cc33", "#00ccff"))
node_sub$color = df$color[match(node_sub$regulatory.function, df$regulatory.function)]
# id
node_sub$id = 1:nrow(node_sub)
# size
node_sub$size  = 3
range(node_sub$size)
# axis
temp = data.frame(mod_col = sel_col, axis = 1:length(sel_col)); temp
node_sub$axis = temp$axis[match(node_sub$regulatory.function, temp$mod)]

# load WGCNA edge file
path = paste0(getwd(), "/Data/WGCNA/edge.txt") # 335845
edge_W = read_tsv(path)[, 1:3] 
# radius
connections = lapply(node_sub$lab, function(s){
  ind_c = c(which(edge_W$fromNode == s), which(edge_W$toNode == s)); # using WGCNA edge here
  data.frame("lab" = s, "count" = length(ind_c))
}) %>% bind_rows()
node_sub = merge(node_sub, connections, by = "lab")
radius = lapply(sel_col, function(s){
  ind = which(node_sub$regulatory.function == s);
  # df = arrange(node_sub[ind, ], desc(count))
  df = arrange(node_sub[ind, ], desc(lab))
  df$radius = seq(20, 200, by = 25)[1:length(ind)]
  df
}) %>% bind_rows()

# node_out
ind = sapply(c("id", "lab", "axis", "radius", "size", "color"), function(s){
  grep(s, names(radius))
})
node_out = radius[, ind]

## manuelly making EDGE
edge_sub$id1 = node_out$id[match(edge_sub$node1, node_out$lab)]
edge_sub$id2 = node_out$id[match(edge_sub$node2, node_out$lab)]
ind = sapply(c("id1", "id2", "combined_score"), function(s){
  grep(s, names(edge_sub))
})
edge_out = edge_sub[, ind] 
names(edge_out) = gsub("combined_score", "weight", names(edge_out))
edge_out$weight = scales::rescale(edge_out$weight, c(0.5, 2))
edge_out$color = "lightgray" 
edge_out = arrange(edge_out, desc(color)) %>% as.data.frame() # first color will be at the bottom
str(edge_out)

# manuelly creating HPD
H1 = list()
H1$nodes = node_out
H1$edges = edge_out
H1$type = "2D"
H1$desc = "cytoscape"
H1$axis.cols = rep('darkgray', length(sel_col)) # make invisible
H1$axLabs = sel_col
class(H1) = "HivePlotData"
chkHPD(H1, confirm = TRUE)
# sumHPD(H1)

anNodes_path = paste0(getwd(), "/Data/WGCNA/HiveR_anNodes_temp.tsv")
df = data.frame("node.lab" = node_out$lab)
df$node.text = node_out$lab
df$angle = 0
df$radius = 0
df$offset = 0
df$hjust = 0.5
df$vjust = 0.5
# write_csv(df, file = anNodes_path)

plotHive(H1, axLabs = sel_col, ch = 5, bkgnd = "transparent",
         axLab.pos = 20, 
         anNodes = anNodes_path,
         anNode.gpar = gpar(col = "black", fontsize = 6, lwd = 0), 
         axLab.gpar = gpar(col = "black", fontsize = 6) ) 

```

